parquet-common/src/main/java/org/apache/parquet/hadoop/metadata/ColumnPath.java

```diff
@@ -83,6 +83,6 @@ public int size() {
   }
 
```

Comment:
I think it would also help to have a `boolean equals(String[] path)` method so that we can make path comparisons without incurring an unnecessary array copy.

Reply:
I know it could be useful for Parquet users such as Trino, but was hesitant to add this as not directly related to immutability. Since toArray became copying, i added toList which avoids copy. Maybe this new method also satisfies those needs and we don't need yet another one?

Reply:
ya, i think that new method will help